great_expectations/datasource/fluent/data_asset/path/directory_asset.py

```diff
@@ -120,3 +175,54 @@ def build_batch_request(
             batch_slice=batch_slice,
             partitioner=partitioner,
         )
+
+    @override
+    def _batch_spec_options_from_batch_request(self, batch_request: BatchRequest) -> dict:
+        """Build a set of options for use in a batch spec from a batch request.
+
+        Args:
+            batch_request: Batch request to use to generate options.
+
+        Returns:
+            Dictionary containing batch spec options.
+        """
+        get_reader_options_include: set[str] | None = self._get_reader_options_include()
+        if not get_reader_options_include:
+            # Set to None if empty set to include any additional `extra_kwargs` passed to `add_*_asset`  # noqa: E501
+            get_reader_options_include = None
+        batch_spec_options = {
+            "reader_method": self._get_reader_method(),
+            "reader_options": self.dict(
+                include=get_reader_options_include,
+                exclude=self._EXCLUDE_FROM_READER_OPTIONS,
+                exclude_unset=True,
+                by_alias=True,
+                config_provider=self._datasource._config_provider,
+            ),
+        }
+
+        self._add_partitioner_batch_parameters(
+            batch_request=batch_request, parameters=batch_spec_options
+        )
+
+        return batch_spec_options
+
+    def _add_partitioner_batch_parameters(self, batch_request, parameters) -> dict:
```

Comment:
Can you add types to the params here? Also non-blocking, but pure functions tend to be simpler to reason about; I'd consider just returning the dict containing `partitioner_method` and `partitioner_kwargs` and merging in the caller.

Reply:
wild, great catch, i thought types were a hard requirement on new code. And love the suggestion of not mutating the dict in the helper method, updated in [c432636](https://github.com/great-expectations/great_expectations/pull/9888/commits/c4326368ae480abd7f6f10dee317d5edfa676000).